### DISCUSSION TITLE
Huawei R4850G2 keine CAN Bus Verbindung

### DISCUSSION THREAD

**Discussion by Phantomias2006** [hoylabs/OpenDTU-OnBattery#299] [Q&A] [answered]
Hallo,

ich bekomme leider keine Verbindung zum Huawei. 
Anbei mal die Hardware:
NodeMCU -> https://www.amazon.de/dp/B0BP1QWVMB?psc=1&ref=ppx_yo2ov_dt_b_product_details
MCP2515 -> https://www.amazon.de/dp/B07SYH9BXS?psc=1&ref=ppx_yo2ov_dt_b_product_details

Die Verkabelung habe ich 10x kontrolliert. Pins sollten alle korrekt angeschlossen sein. Auch  die Modifikation auf 5V habe ich durchgeführt. Im GUI habe ich den Huawei aktiviert, per MQTT stehen aber alle Werte auf 0 (bis auf die Verbindungszeit… die zählt nach oben)
MCP2515 Boards habe ich zwei Probiert. 

Fällt euch noch etwas ein wie bzw. was ich testen könnte?
LG
Florian

Edit: Der Jumper für den Endwiderstand ändert nichts an meiner Situation. Aber muss dieser gesetzt sein oder nicht?

**Comment by Phantomias2006** [comment]
![IMG_8051](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/15437240/f2034821-bd2f-4574-8422-401023eeca02)
![IMG_8052](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/15437240/1019a562-33c6-4572-95f4-ad36bf61e11d)
![IMG_8053](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/15437240/408ce114-1627-4b83-a00b-99897c480b2a)
![IMG_8054](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/15437240/b75e63d0-63f1-4252-91ed-958013560cbd)

**Comment by Manos1966** [comment]
Wo ist das "Huawei Slot Detect" dass oben rechts auf das Bild (siehe unten) zu sehen ist?
https://github.com/helgeerbe/OpenDTU-OnBattery/blob/master/docs/Wiring_ESP32_Symbol.png

**Comment by Phantomias2006** [comment]
Hi,

danke für deine Antwort.
Slot Detect 1 & 2 habe ich auf Masse angeschlossen. Soweit ich weiß dient es nur dazu um das Netzteil zu aktivieren. Dadurch ist es jetzt dauerhaft aktiviert. Netzteil selber läuft auch an und Liefert Spannung. Nur die CAN Verbindung klappt nicht.

LG

**Comment by bf4ed** [comment]
Bei Device-Manager hast du da ein profil ausgewählt?
Und wenn, ist pin mapping korrekt?

Bei mir funktioniert der Huawei CAN bus.
Aber hab das problem mit Pylontech CAN bus. ah... :)

**Comment by Phantomias2006** [comment, marked as answer]
Hi, 
Board/Profil war ausgewählt.
Habe mir mal die Debug ausgaben angesehen und gemerkt dass das Pin Mapping einen error wirft.
Es sieht so aus als würde das Mapping für nodeMCU komplett fehlen….
Sollte das nicht in der Platformio.ini bei den Boards gesetzt werden? Bei ein paar ist es vorhanden und bei einigen aber nicht.
Zumindest läuft es jetzt.
Danke für deine Hilfe :)
LG

**Comment by Phantomias2006** [comment]
Jetzt muss ich doch noch einmal nachfragen. 
Grundsätzlich funktioniert ja jetzt die CAN anbindung an den Huawei, bei längerer Laufzeit jedoch kann das Netzteil nur mehr sporadisch ausgelesen werden. Sobald der ESP neu gestartet wird klappt es wieder für ~1h
Ich habe date_age mal aufgezeichnet und visualisiert damit ihr seht was ich meine. Im Chart seht ihr auch immer schön wann ich einen reboot gemacht habe. Gibt es hier einen Software bug?

**Reply by CaCu15** [reply]
Hi, das Problem habe ich mit den neueren OpenDTU onBattery Releases auch. Ich bin mit einem Release von Anfang Mai gestartet, da gab es diesen Effekt nicht, man hatte IMMER aktuelle Daten vom Huawei. Mit den Releases ab Juni habe ich auch häufig fehlende Daten vom Huawei. Das Senden von neuen Limits an das Huawei funktioniert problemlos, aber die Daten im der Liveview sind häufig nicht aktuell.

Vielleicht magst Du auch mal mit einem Release von Anfang Mai testen? Wenn es damit funktioniert, dann weißt Du wenigstens, dass es NICHT an der HW liegt.

**Reply by sabafreiburg** [reply]
Ich sehe auch eine instabile Verbingung zum Huawei.  Ich habe kein Logging laufen, aber wenn man die Live Ansicht aus dem Augenwinkel beobachtet, ist das Huawei häufig rot.  Für  30sec .... 1...2...3...4 Minuten und dann ist wieder für eine ganze Weile ok. Man sieht dann z.B. bei schwankendem Stromverbrauch oder wolkigem Himmel auch, wie minutenlang Überschuss ins Netz geht.
Nach was könnte ich denn mal in der Konsole ausschau halten?

**Comment by Phantomias2006** [comment]
![huawei](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/15437240/4ac2a0c2-25aa-437c-8db8-d690500046b0)

**Comment by erwano1957** [comment]
Hallo, ich nutze auch eine Huawei und steuere ihn über den CAN Anschluss. Ich bin dabei diesem Video von KlausI gefolgt und es funktioniert prima und ohne Unterbrechung. Bitte vergleiche es doch mal mit einem Ansatz.  Speziell das Durchtrennen der Leiterbahn kann ich auf deinem Photo nicht erkennen.  https://www.youtube.com/watch?v=2mAzP_35Ox8

**Comment by Manos1966** [comment]
**FRAGE:** Hat jemand hier den HUAWAI R485G2 mit dem SN65HVD230-CAN-Board probiert?

![Preis SN65HVD230-CAN-Board](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/17d334a0-650f-4c99-8b3b-ff5937e12797)

**Reply by CaCu15** [reply]
Hallo Manos1966, 
 das kann m.E. auch nicht funktionieren, siehe auch [https://github.com/helgeerbe/OpenDTU-OnBattery/blob/development/docs/hardware_flash.md](https://github.com/helgeerbe/OpenDTU-OnBattery/blob/development/docs/hardware_flash.md)
 
Dieses Board hat nur einen CAN Transceiver, aber keinen CAN Controller. Der auf dem ESP32 integrierte CAN Controller wird für die Kommunikation mit dem Akku verwendet. Da das Huawei einen CAN mit anderen Parameter (Übertragungsrate) benötigt,  brauchst Du für die Kommunikation mit dem Huawei einen separaten Controller. Der ist hier nicht enthalten, darum kann das m.E. nicht funktionieren. 
 
Der Transceiver ist für die physische Anbindung des CAN Busses zuständig (OSI Layer 1), also die Umwandlung der Nachrichten in elektrische Impulse, die über die beiden Adern übertragen werden. 
Der Controller macht die Abwicklung des CAN Protocolls (OSI Layer 2), also die Bildung der Datenpakete inkl. Prüfsummen etc., die übertragen werden soll. 

 Siehe Erklärung z.B. auf  z.B. [https://www.analog.com/en/technical-articles/understanding-can-transceiver-how-validate-multinode-can-system-performance.html](https://www.analog.com/en/technical-articles/understanding-can-transceiver-how-validate-multinode-can-system-performance.html)

**Reply by Manos1966** [reply]
Vielen Dank @CaCu15 fuer die Erklaerung.

Jetzt verstehe ich endlich den Unterschied und warum ich diese Platine brauche 👍 
![Youmile 2 Stk MCP2515 CAN-Bus-Modul](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/72de1348-2ee9-439d-a2a0-39d7c362a77a)

**Comment by RobertElektro** [comment]
Cooles Projekt mit sehr guten Erklärungen. Nutze für all meine anderen Steuerungen ESP8266 (Wemos D1 Mini). Deshalb frage ich mich, funktioniert das Setting wohl auch mit so einem MCP2515 D1 Mini CAN-Bus Shield, z. B. von kleinanzeigen.de ? Wegen der Spannungen?

Gruß, Robert


![CAN-MCP2515](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/154672021/0c91ce84-ff7f-46a9-b229-94a3d7a04f8e)

**Reply by madmartin** [reply]
Hallo Robert,

der ESP8266 hat ebenso wie der ESP32 auf den GPIOs 3,3 Volt Pegel. Wenn dieses "Shield" mit dem ESP8266 funktioniert, sollte es rein elektrisch mit dem ESP32 auch funktionieren. Mechanisch huckepack zusammenpassen wird es aber wohl nicht.

**Reply by RobertElektro** [reply]
Hallo Martin, vielen Dank für die schnelle Antwort. Ich frag evtl. mal nach ob man auch einen ESP32 D1 Mini draufstecken kann.

**Reply by Manos1966** [reply]
Hallo Leute,

auch hier ein Auge werfen:
[CANbus communication with OpenDTU](https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Huawei-AC-PSU#canbus-communication-with-opendtu)

(Joy-IT SBC-CAN01 Platine)

**Reply by RobertElektro** [reply]
Moin, hab mich schlauf gemacht, laut Aussagen des Verkäufers passt sowohl das Wemos Di Mini ESP8266 als auch das ESP32 drauf.
